### PR TITLE
add github actions

### PR DESCRIPTION
**Pull request by N-R-K**
this runs both default and minimal build with gcc and clang.
tcc seems to fail at link time for some reason.

---

I'm not familiar with github actions/workflow, simply hacked this together from the provided template. Seems to be working fine, here's some example:

* https://github.com/N-R-K/nsxiv/runs/4614439272?check_suite_focus=true <br> (minimal build failing due to `EXIT_SUCCESS` not being defined)
* https://github.com/N-R-K/nsxiv/pull/2 <br> (PR fixing it)

### CODE REVIEW

**Comment by N-R-K** [comment]
@TAAPArthur Any clue on why tcc is failing at linking?

**Comment by TAAPArthur** [comment]
> @TAAPArthur Any clue on why tcc is failing at linking?

I'll take a look. Without looking, I'd guess because it is not to using a recent tcc version. The last "stable" release is really old and behind the git version.

**Comment by TAAPArthur** [comment]
Yeah it looks like 0.9.27-8 is used. Probably should just build tcc from mob (or some recent hash) if we want to use tcc.

Maybe something like
```
git clone git://repo.or.cz/tinycc.git
( cd tinycc && ./configure && make)
CC=$PWD/tinycc/tcc OPT_DEP_DEFAULT=1 make
```

Or of course we could find some binary somewhere.

Can confirm that tcc can build nsixv

**Comment by N-R-K** [comment]
Now it fails compilation and says something about not finding `stdarg.h`. I'm assuming it's environment related issue: https://github.com/N-R-K/nsxiv/runs/4613227345?check_suite_focus=true

**Comment by N-R-K** [comment]
Wonder if someone here is familiar with github actions and can take a look into this for best practices and whatnot.
@kdkasad Giving a ping since I recall you've mentioned github actions somewhere before.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: make
+      run: |
+        sudo apt-get install libimlib2 libimlib2-dev \
+                             xserver-xorg-core xserver-xorg-dev \
+                             libxft2 libxft-dev libexif12 libexif-dev \
+                             gcc clang >/dev/null
+        git clone 'git://repo.or.cz/tinycc.git' >/dev/null
```
Comment (kdkasad) [inline .github/workflows/build.yml:23]: It would be cleaner to use `git clone --quiet ...`

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: make
+      run: |
+        sudo apt-get install libimlib2 libimlib2-dev \
+                             xserver-xorg-core xserver-xorg-dev \
+                             gcc clang >/dev/null
+        sudo apt-get remove libxft2 libxft-dev libexif12 libexif-dev >/dev/null
+        git clone 'git://repo.or.cz/tinycc.git' >/dev/null
```
Comment (kdkasad) [inline .github/workflows/build.yml:42]: `git clone --quiet` works here too

**Review by kdkasad — COMMENTED** [review]
I'm not very well-versed with GitHub Actions but I would assume that the building/installing TinyCC could be separated into a different step from that of building nsxiv.

It seems like that could even be extracted to a separate job, so that the building of TinyCC only happens once, then the `full-build` and `minimal-build` jobs run in parallel after that completes.

So far I've just added some comments on usage of commands, not the structure of the actions themselves.

**Comment by N-R-K** [comment]
@kdkasad Thanks for the review, I've pushed the `--quiet` change.

As for the cache, I'll look into it in a bit. But without looking, on our `minimal-build` we're explicitly uninstalling some optional deps to make sure everything compiles/links fine without the optional headers/deps being installed. So that might require some extra fiddling with the cache to get it to work, I presume.

**Comment by N-R-K** [comment]
Thinking again, since our build process is rather fast (25 seconds) I'm fine leaving things be and not bother with caching.

**Comment by N-R-K** [comment]
Unless someone wants to bother with caching, I think this PR should be ready to be merged.

**Review by TAAPArthur — APPROVED** [review]

**Review by mamg22 — APPROVED** [review]

**Comment by mamg22** [comment]
[The first build failed](https://github.com/nsxiv/nsxiv/actions/runs/1631926746), failing when executing `actions/checkout@v2`. It seems to be an issue with the action settings in the repo limiting what actions can be run, as indicated [in this question](https://github.community/t/checkout-v2-error-actions-in-this-workflow-must-within-a-repository-owned-by-organization/136315/2).

**Comment by N-R-K** [comment]
I've enabled "allow all actions" in both the repo and the org. The workflow seems to be running now, though it failed at cloning tcc. https://github.com/nsxiv/nsxiv/pull/194

**Comment by N-R-K** [comment]
The workflow fails on my repo as well,

Seems like the Tcc repo isn't reliable. I've switched to a github mirror, cleaned up the commit msg and forced push the changes.

**Comment by N-R-K** [comment]
Now everything is working as expected: https://github.com/nsxiv/nsxiv/runs/4658052056?check_suite_focus=true

The minimal build fails, but that's expected due to #194

**Comment by mamg22** [comment]
Regarding re-triggering checks, there's a button in the checks tab of a PR that allows running all jobs again, which seems better than closing and reopening.

**Comment by N-R-K** [comment]
> there's a button in the checks tab of a PR that allows running all jobs again

That only appears when a workflow has failed afaik. The workflow never ran on that PR to begin with, so had to close and reopen.

**Pull request merged**